### DISCUSSION TITLE
Is `ccache --config-path ccache.conf g++` supported?

### DISCUSSION THREAD

**Discussion by dyadav7** [ccache/ccache#990] [Questions & answers] [answered]
is "ccache --config-path ccache.conf g++ ..." not supported ?

```
z 18:54 ~/cpp $ ccache --config-path /tmp/ccache.conf g++wp-5.4.0 -m64 -std=c++11 -c lock.cxx
ccache: invalid option -- 'm'
z 18:55 ~/cpp $ ls  lock.o
ls: cannot access lock.o: No such file or directory
z 18:55 ~/cpp $ env CCACHE_CONFIGPATH=/tmp/ccache.conf ccache g++wp-5.4.0 -m64 -std=c++11 -c lock.cxx
z 18:55 ~/cpp $ ls lock.o
lock.o
z 18:55 ~/cpp $ 
```

**Comment by jrosdahl** [comment, marked as answer]
No. When run in compiler mode, ccache does not parse any options of its own (with `--ccache-skip` being an exception). See [_Synopsis_](https://ccache.dev/manual/4.5.1.html#_synopsis) for the different ways to execute ccache.

I'm starting to think that it was a mistake to add the `--config-path` option since it apparently contributes with confusion.

**Reply by dyadav7** [reply]
Currently i use "env CCACHE_CONFIGPATH=/tmp/ccache.conf ...  ccache  g++  ..." command.
It would be better if the need to use env/setenv command is not there:
`     ccache  --config-path /tmp/ccache.conf g++ ...`

i think the above should be supported and i guess the config file has ways to enable/disable every ccache setting.
This will avoid the need of using **env / setenv/ export** command for using ccache.

**Reply by jrosdahl** [reply]
You don't have to use `env` if you run the command via the shell:

    CCACHE_CONFIGPATH=/tmp/ccache.conf ... ccache g++ ...

I don't quite understand your use case. Why do you want to generate a temporary config file? Can't you just set the variables you need in the environment, like

    CCACHE_AAA=BBB ccache g++ ...

?

**Reply by dyadav7** [reply]
I use csh. i can not use` CCACHE_CONFIGPATH=/tmp/ccache.conf ... ccache g++`
I have to use env command to temporary set the CCACHE_CONFIGPATH.

I do not use temporary config file. 

For my majority ccache usage - i would like to use :
` ccache --config-path /tmp/ccache.conf g++ ...`

My ccache confg file is fixed - i change it rarely.

I there is some setting i want to enable temporarlly then i am ok with using env/setenv CCACHE_..  to set it.
So, only for exceptional cases i would like to use env/setenv/CCAHCE_xx= commands. I am ok with using this style for such corner cases. 

If config files is fixed and one does not use other CCACHE_.. env settings - then one need not even use the CCACHE_CONFIGPATH= to specify the config path only.

I am using only one env CCACHE_CONFIGPATH.
I think the ccache command line will be become simpler. The dependency on env/CCAHE_=/ will not be there for most of cases.
    `ccache --config-path /tmp/ccache.conf  <Compile Command>`

**Reply by jrosdahl** [reply]
> I do not use temporary config file.

Ah. How about placing the config file in the cache directory (where it's designed to be located)?

**Reply by dyadav7** [reply]
my configuration is like this :
```
  Cache directory:      /s/misc/ccache/db
  Primary config:       /s/misc/ccache/conf/ccache.conf
```

**Reply by jrosdahl** [reply]
Ah. How about placing the config file in the cache directory (where it's designed to be located)? :wink:

**Reply by dyadav7** [reply]
you mean moving /s/misc/ccache/conf/ccache.conf to /s/misc/ccache/db/ccache.conf ?
wil this avoid the need of adding _`"cache_dir = /s/zinqa/ccache/db"`_ line in ccache.conf ?

**Reply by jrosdahl** [reply]
Yes. That's how it's supposed to work: you configure the cache directory location (or use the default one), either by setting `CCACHE_DIR` or by setting `cache_dir` in the secondary configuration file, typically `/usr/local/etc/ccache.conf` (or whatever location you choose for the secondary config when building ccache). The cache directory then contains both configuration, temporary directory, file storage and statistics counters.
